File: benchmarks/tsb/bench_dataframe_rolling.ts
/**
 * Benchmark: DataFrameRolling mean on 100k-row DataFrame
 */
import { DataFrame } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const a = Array.from({ length: ROWS }, (_, i) => i * 0.1);
const b = Array.from({ length: ROWS }, (_, i) => i * 0.2);
const df = DataFrame.fromColumns({ a, b });

for (let i = 0; i < WARMUP; i++) df.rolling(10).mean();
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) df.rolling(10).mean();
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "dataframe_rolling",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_dataframe_rolling.py
"""Benchmark: DataFrame rolling mean on 100k-row DataFrame"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
df = pd.DataFrame({"a": [i * 0.1 for i in range(ROWS)], "b": [i * 0.2 for i in range(ROWS)]})

for _ in range(WARMUP):
    df.rolling(10).mean()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.rolling(10).mean()
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "dataframe_rolling", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
